src/app/shared/services/courses.service.ts

```diff
@@ -34,12 +35,20 @@ export class CoursesService {
     return this.httpCourses.getCourse(id);
   }
 
-  getSearchedCourses(searchTerm: string) {
-    const requestBody = {
-      textFragment: searchTerm,
-      sort: 'date'
-    }
-    return this.httpCourses.getCourses(requestBody);
+  getSearchedCourses(searchTerm: string): Observable<Course[]> {
+    this.searchSubject$.next(searchTerm);
+
+    return this.searchSubject$.pipe(
+      distinctUntilChanged(),
+      debounceTime(500),
```

Comment:
just call the method here. You need to denounce the search emit in the search component